plugins/syncbeacon/plugin.go

```diff
@@ -0,0 +1,217 @@
+	CfgSyncBeaconCleanupInterval = "syncbeacon.cleanupInterval"
+
+	// The percentage of following nodes that have to be synced.
+	syncPercentage = 0.6
+)
+
+// beaconSync represents the beacon payload and the sync status.
+type beaconSync struct {
+	payload *Payload
+	synced  bool
+}
+
+func init() {
+	flag.StringSlice(CfgSyncBeaconFollowNodes, []string{"Gm7W191NDnqyF7KJycZqK7V6ENLwqxTwoKQN4SmpkB24", "9DB3j9cWYSuEEtkvanrzqkzCQMdH1FGv3TawJdVbDxkd"}, "list of trusted nodes to follow their sync status")
+	flag.Bool(CfgSyncBeaconPrimary, false, "if this node should be a sync beacon")
+	flag.Int(CfgSyncBeaconBroadcastIntervalSec, 30, "the interval at which the node will broadcast ist sync status")
+	flag.Int(CfgSyncBeaconMaxTimeWindowSec, 10, "the maximum time window for which a sync payload would be considerable")
+	flag.Int(CfgSyncBeaconMaxTimeOfflineSec, 40, "the maximum time the node should stay without receiving updates")
+	flag.Int(CfgSyncBeaconCleanupInterval, 10, "the interval at which cleanups are done")
+}
+
+var (
+	// plugin is the plugin instance of the sync beacon plugin.
+	plugin                *node.Plugin
+	once                  goSync.Once
+	log                   *logger.Logger
+	beaconSyncMap         map[string]beaconSync
+	beaconNodesPublicKeys []string
+)
+
+// Plugin gets the plugin instance.
+func Plugin() *node.Plugin {
+	once.Do(func() {
+		plugin = node.NewPlugin(PluginName, node.Disabled, configure, run)
+	})
+	return plugin
+}
+
+// configure events
+func configure(_ *node.Plugin) {
+	log = logger.NewLogger(PluginName)
+	beaconNodesPublicKeys = config.Node().GetStringSlice(CfgSyncBeaconFollowNodes)
+	beaconSyncMap = make(map[string]beaconSync, len(beaconNodesPublicKeys))
+	for _, pubKey := range beaconNodesPublicKeys {
+		beaconSyncMap[pubKey] = beaconSync{
+			payload: nil,
+			synced:  false,
+		}
+	}
+
+	messagelayer.Tangle().Events.MessageSolid.Attach(events.NewClosure(func(cachedMessage *message.CachedMessage, cachedMessageMetadata *tangle.CachedMessageMetadata) {
+		cachedMessageMetadata.Release()
+		cachedMessage.Consume(func(msg *message.Message) {
+			payload := &Payload{}
+			err := payload.Unmarshal(msg.Payload().Bytes())
+			if err != nil {
+				return
+			}
+			if !IsSyncBeaconPayload(payload) {
+				return
+			}
+
+			handlePayload(payload, msg.IssuerPublicKey())
+		})
+	}))
+}
+
+// handlePayload handles the received payload. It does the following checks:
+// It checks that the issuer of the payload is a followed node.
+// The time that payload was sent is not greater than  CfgSyncBeaconMaxTimeWindowSec. If the duration is longer than CfgSyncBeaconMaxTimeWindowSec, we consider that beacon to be out of sync till we receive a newer payload.
+// More than syncPercentage of followed nodes are also synced, the node is set to synced. Otherwise, its set as desynced.
+func handlePayload(syncBeaconPayload *Payload, issuerPublicKey ed25519.PublicKey) {
+	issuerPublicKeyStr := issuerPublicKey.String()
```

Comment:
You can save this conversion and avoid the following loop as I suggested in the other comment.

There's no real need for `beaconNodesPublicKeys` as a slice. You could create a `map[string][]byte` mapping the public key from string to its byte representation and then directly compare the bytes. As this map is only created/modified on startup it doesn't need to be locked on access. So the time holding a lock can be minimized.